### ISSUE TITLE
some errors i getting

### ISSUE THREAD

**Issue by comet424** [marcolivierarsenault/moonraker-home-assistant#370]
while i still printing so i cant test that other issue with the printer name issue i was looking at logs.. and see  i got custom moonraker errors
here are the errors i getting

if you need more let me know  
and i buggered up typing in between the lines so it wasnt posting right and i pasted in the logs below everything  

## Describe the bug

bug just errors  i reporting

## Moonraker Logs

Logger: custom_components.moonraker
Source: custom_components/moonraker/__init__.py:222
integration: Moonraker ([documentation](https://moonraker-home-assistant.readthedocs.io/en/latest/), [issues](https://github.com/marcolivierarsenault/moonraker-home-assistant/issues))
First occurred: 1:17:15 PM (20 occurrences)
Last logged: 1:18:19 PM

gcode {{'size': 19029014, 'modified': 1721305144.233158, 'print_start_time': 1721306963.4750283, 'job_id': '00000E', 'filename': 'Remote controls holder_0.2mm_PETG_Elegoo Neptune 4 Max_5h14m.gcode'}}
gcode {{'size': 12923364, 'modified': 1721304963.5036504, 'print_start_time': 1721306978.5057268, 'job_id': '000017', 'filename': 'Remote Holder_0.2mm_PETG_Elegoo Neptune 4 Max_6h41m.gcode'}}
```
```
This error originated from a custom integration.
Logger: custom_components.moonraker
Source: custom_components/moonraker/__init__.py:221
integration: Moonraker ([documentation](https://moonraker-home-assistant.readthedocs.io/en/latest/), [issues](https://github.com/marcolivierarsenault/moonraker-home-assistant/issues))
First occurred: 1:17:15 PM (20 occurrences)
Last logged: 1:18:19 PM

Query Object {{'filename': 'Remote controls holder_0.2mm_PETG_Elegoo Neptune 4 Max_5h14m.gcode'}}
Query Object {{'filename': 'Remote Holder_0.2mm_PETG_Elegoo Neptune 4 Max_6h41m.gcode'}}
```
```
This error originated from a custom integration.
Logger: custom_components.moonraker
Source: custom_components/moonraker/__init__.py:220
integration: Moonraker ([documentation](https://moonraker-home-assistant.readthedocs.io/en/latest/), [issues](https://github.com/marcolivierarsenault/moonraker-home-assistant/issues))
First occurred: 1:17:15 PM (20 occurrences)
Last logged: 1:18:19 PM

failed to get thumbnails {'thumbnails'}
```
i got these errors when i rebooted home assistant  so im sure they were there before  but i didnt check


some of the extra logs are 
```
2024-07-18 13:17:18.048 ERROR (MainThread) [custom_components.moonraker] failed to get thumbnails {'thumbnails'}
2024-07-18 13:17:18.048 ERROR (MainThread) [custom_components.moonraker] Query Object {{'filename': 'Remote controls holder_0.2mm_PETG_Elegoo Neptune 4 Max_5h14m.gcode'}}
2024-07-18 13:17:18.048 ERROR (MainThread) [custom_components.moonraker] gcode {{'size': 19029014, 'modified': 1721305144.233158, 'print_start_time': 1721306963.4750283, 'job_id': '00000E', 'filename': 'Remote controls holder_0.2mm_PETG_Elegoo Neptune 4 Max_5h14m.gcode'}}
2024-07-18 13:17:18.048 ERROR (MainThread) [custom_components.moonraker] failed to get thumbnails {'thumbnails'}
2024-07-18 13:17:18.048 ERROR (MainThread) [custom_components.moonraker] Query Object {{'filename': 'Remote controls holder_0.2mm_PETG_Elegoo Neptune 4 Max_5h14m.gcode'}}
2024-07-18 13:17:18.048 ERROR (MainThread) [custom_components.moonraker] gcode {{'size': 19029014, 'modified': 1721305144.233158, 'print_start_time': 1721306963.4750283, 'job_id': '00000E', 'filename': 'Remote controls holder_0.2mm_PETG_Elegoo Neptune 4 Max_5h14m.gcode'}}

2024-07-18 13:17:18.229 ERROR (MainThread) [custom_components.moonraker] failed to get thumbnails {'thumbnails'}
2024-07-18 13:17:18.230 ERROR (MainThread) [custom_components.moonraker] Query Object {{'filename': 'Remote Holder_0.2mm_PETG_Elegoo Neptune 4 Max_6h41m.gcode'}}
2024-07-18 13:17:18.230 ERROR (MainThread) [custom_components.moonraker] gcode {{'size': 12923364, 'modified': 1721304963.5036504, 'print_start_time': 1721306978.5057268, 'job_id': '000017', 'filename': 'Remote Holder_0.2mm_PETG_Elegoo Neptune 4 Max_6h41m.gcode'}}
2024-07-18 13:17:18.234 ERROR (MainThread) [custom_components.moonraker] failed to get thumbnails {'thumbnails'}
2024-07-18 13:17:18.234 ERROR (MainThread) [custom_components.moonraker] Query Object {{'filename': 'Remote Holder_0.2mm_PETG_Elegoo Neptune 4 Max_6h41m.gcode'}}
2024-07-18 13:17:18.234 ERROR (MainThread) [custom_components.moonraker] gcode {{'size': 12923364, 'modified': 1721304963.5036504, 'print_start_time': 1721306978.5057268, 'job_id': '000017', 'filename': 'Remote Holder_0.2mm_PETG_Elegoo Neptune 4 Max_6h41m.gcode'}}
2024-07-18 13:17:18.289 ERROR (MainThread) [custom_components.moonraker] failed to get thumbnails {'thumbnails'}
2024-07-18 13:17:18.289 ERROR (MainThread) [custom_components.moonraker] Query Object {{'filename': 'Remote controls holder_0.2mm_PETG_Elegoo Neptune 4 Max_5h14m.gcode'}}
2024-07-18 13:17:18.289 ERROR (MainThread) [custom_components.moonraker] gcode {{'size': 19029014, 'modified': 1721305144.233158, 'print_start_time': 1721306963.4750283, 'job_id': '00000E', 'filename': 'Remote controls holder_0.2mm_PETG_Elegoo Neptune 4 Max_5h14m.gcode'}}
2024-07-18 13:17:18.293 ERROR (MainThread) [custom_components.moonraker] failed to get thumbnails {'thumbnails'}
2024-07-18 13:17:18.293 ERROR (MainThread) [custom_components.moonraker] Query Object {{'filename': 'Remote controls holder_0.2mm_PETG_Elegoo Neptune 4 Max_5h14m.gcode'}}
2024-07-18 13:17:18.293 ERROR (MainThread) [custom_components.moonraker] gcode {{'size': 19029014, 'modified': 1721305144.233158, 'print_start_time': 1721306963.4750283, 'job_id': '00000E', 'filename': 'Remote controls holder_0.2mm_PETG_Elegoo Neptune 4 Max_5h14m.gcode'}}
2024-07-18 13:17:18.488 ERROR (MainThread) [custom_components.moonraker] failed to get thumbnails {'thumbnails'}
2024-07-18 13:17:18.488 ERROR (MainThread) [custom_components.moonraker] Query Object {{'filename': 'Remote Holder_0.2mm_PETG_Elegoo Neptune 4 Max_6h41m.gcode'}}
2024-07-18 13:17:18.488 ERROR (MainThread) [custom_components.moonraker] gcode {{'size': 12923364, 'modified': 1721304963.5036504, 'print_start_time': 1721306978.5057268, 'job_id': '000017', 'filename': 'Remote Holder_0.2mm_PETG_Elegoo Neptune 4 Max_6h41m.gcode'}}
2024-07-18 13:17:18.490 ERROR (MainThread) [custom_components.moonraker] failed to get thumbnails {'thumbnails'}
2024-07-18 13:17:18.490 ERROR (MainThread) [custom_components.moonraker] Query Object {{'filename': 'Remote Holder_0.2mm_PETG_Elegoo Neptune 4 Max_6h41m.gcode'}}
2024-07-18 13:17:18.490 ERROR (MainThread) [custom_components.moonraker] gcode {{'size': 12923364, 'modified': 1721304963.5036504, 'print_start_time': 1721306978.5057268, 'job_id': '000017', 'filename': 'Remote Holder_0.2mm_PETG_Elegoo Neptune 4 Max_6h41m.gcode'}}
2024-07-18 13:17:18.556 ERROR (MainThread) [custom_components.moonraker] failed to get thumbnails {'thumbnails'}
2024-07-18 13:17:18.556 ERROR (MainThread) [custom_components.moonraker] Query Object {{'filename': 'Remote controls holder_0.2mm_PETG_Elegoo Neptune 4 Max_5h14m.gcode'}}
2024-07-18 13:17:18.556 ERROR (MainThread) [custom_components.moonraker] gcode {{'size': 19029014, 'modified': 1721305144.233158, 'print_start_time': 1721306963.4750283, 'job_id': '00000E', 'filename': 'Remote controls holder_0.2mm_PETG_Elegoo Neptune 4 Max_5h14m.gcode'}}
2024-07-18 13:17:18.557 ERROR (MainThread) [custom_components.moonraker] failed to get thumbnails {'thumbnails'}
2024-07-18 13:17:18.557 ERROR (MainThread) [custom_components.moonraker] Query Object {{'filename': 'Remote controls holder_0.2mm_PETG_Elegoo Neptune 4 Max_5h14m.gcode'}}
2024-07-18 13:17:18.557 ERROR (MainThread) [custom_components.moonraker] gcode {{'size': 19029014, 'modified': 1721305144.233158, 'print_start_time': 1721306963.4750283, 'job_id': '00000E', 'filename': 'Remote controls holder_0.2mm_PETG_Elegoo Neptune 4 Max_5h14m.gcode'}}

```
logs above i couldnt post the entire  home assistant logs it was too big
```

**Comment by marcolivierarsenault** [comment]
can you provide a full copy of your log https://moonraker-home-assistant.readthedocs.io/en/stable/support/logs.html#get-a-copy-of-your-ha-logs

**Comment by comet424** [comment]
[home-assistant_2024-08-19T21-23-19.874Z.log](https://github.com/user-attachments/files/16665459/home-assistant_2024-08-19T21-23-19.874Z.log)

**Comment by marcolivierarsenault** [comment]
Looks like your GCode does not include a thumbnail 
`gcode {{'size': 22293291, 'modified': 1724072820.0976112, 'print_start_time': 1724072855.064634, 'job_id': '00001A', 'filename': 'Mousetrap_Main_Body_0.2mm_PETG_Elegoo Neptune 4 Max_10h48m.gcode'}}`

Look at this one for exemple
{"result": {"size": 6572476, "modified": 1721164427.5377321, "uuid": "96a8cc56-ab39-42f1-ae1a-50ce2271f280", "slicer": "Cura", "slicer_version": "5.7.0", "gcode_start_byte": 282, "gcode_end_byte": 6571450, "layer_count": 468, "object_height": 93.6, "estimated_time": 12115.0, "layer_height": 0.2, "first_layer_height": 0.2, "filament_total": 5312.14, "thumbnails": [{"width": 32, "height": 32, "size": 2180, "relative_path": ".thumbs/CE3E3V2_t65_x_wing_print-in-place-32x32.png"}, {"width": 300, "height": 300, "size": 27174, "relative_path": ".thumbs/CE3E3V2_t65_x_wing_print-in-place.png"}], "print_start_time": 1721164434.3014824, "job_id": "00016A", "filename": "CE3E3V2_t65_x_wing_print-in-place.gcode"}}

This is something your slicer should do.

What is your slicer?

**Comment by marcolivierarsenault** [comment]
can you try another slicer. (I know Cura works)

**Comment by comet424** [comment]
i can try  ill find something to print  and get back to ya

**Comment by comet424** [comment]
here is logs when i used  Cura 5.6
[home-assistant_2024-08-20T04-02-39.287Z.log](https://github.com/user-attachments/files/16668835/home-assistant_2024-08-20T04-02-39.287Z.log)
and i get same error with  updated Oraca Slicer  2.1.1

**Comment by marcolivierarsenault** [comment]
Make sure you have enable the thumbnail in cura-moonraker pluggin 

![image](https://github.com/user-attachments/assets/a8f7a492-8319-40a2-9355-f350b7cdbbd4)

**Comment by marcolivierarsenault** [comment]
but TLDR, this is a fix that you have to implement in your slicer (regardless what sensor you select)

**Comment by comet424** [comment]
what tldr?   and if your talking about a picture to show up on the screen on the printer.. then the thumbnail shows up there so you know what your printing..

ok ill try this UFP with thumbnail whatever that is.. and whatever TLDR is

**Comment by marcolivierarsenault** [comment]
TLDR = Too long did not read.

But you can interpret this as, 
`in conclusion, this is a fix that you have to implement in your slicer (regardless what sensor you select)`

**Comment by tobiasbayer** [comment]
I'd suggest changing the log level to `WARN` or even `INFO` or `DEBUG`. A gcode file without a thumbnail prints perfectly fine. I don't put thumbnails in my gcode file either and get hammered with error messages in HA's log then as well.

**Comment by marcolivierarsenault** [comment]
I like that idea. Will move to Warning
